### ISSUE TITLE
cmd module tab misbehavior

### ISSUE THREAD

**Issue by b055b20d-e7de-4716-83a6-fe8742346ad5** [python/cpython#53279]
BPO | [9033](https://bugs.python.org/issue9033)
--- | :---
Nosy | @ronaldoussoren, @ned-deily, @merwok, @zvezdan, @boompig
Files | <li>[cmd.patch](https://bugs.python.org/file17782/cmd.patch "Uploaded as text/plain at 2010-06-27.14:33:33 by @ronaldoussoren")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/ronaldoussoren'
closed_at = None
created_at = <Date 2010-06-19.15:10:14.122>
labels = ['OS-mac', 'type-bug']
title = 'cmd module tab misbehavior'
updated_at = <Date 2018-01-17.22:24:11.948>
user = 'https://bugs.python.org/slcott'
```

bugs.python.org fields:
```python
activity = <Date 2018-01-17.22:24:11.948>
actor = 'boompig'
assignee = 'ronaldoussoren'
closed = False
closed_date = None
closer = None
components = ['macOS']
creation = <Date 2010-06-19.15:10:14.122>
creator = 'slcott'
dependencies = []
files = ['17782']
hgrepos = []
issue_num = 9033
keywords = ['patch']
message_count = 13.0
messages = ['108185', '108196', '108200', '108353', '108379', '108380', '108381', '108392', '108792', '109732', '109746', '193157', '310213']
nosy_count = 7.0
nosy_names = ['ronaldoussoren', 'ned.deily', 'eric.araujo', 'zvezdan', 'l0nwlf', 'slcott', 'boompig']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue9033'
versions = ['Python 3.6']
```

</p></details>

**Comment by ronaldoussoren** [comment]
Issue is still present, although command completion does work in an interactive shell (that is, start ``cmdloop()`` from the REPL. 

Also: site.py contains code that similar to my "cmd.patch", I intend to create a PR based on it.

**Comment by encukou** [comment]
This was fixed in #107748, for Python 3.13. From my Mac:
```
$ PYTHON_BASIC_REPL=1 ./python.exe
Python 3.14.0a0 (heads/main:2c7209a3bdf, Jun 17 2024, 17:10:56) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> i<tab><tab>
id(         if          import      in          input(      int(        is          isinstance( issubclass( iter(      
>>> import readline
>>> readline.<tab><tab>
readline.add_history(                         readline.read_history_file(                  
readline.backend                              readline.read_init_file(                     
readline.clear_history()                      readline.redisplay()                         
[... several lines omitted ...]                  
>>> readline.backend
'editline'

```

It's a new feature, so it's not backported. But, in Python 3.11 tab completion should work if you install readline and build with `./configure --with-readline=readline`.
